### PR TITLE
add event sources yaml

### PR DESCRIPTION
**Pull request by MohamedFarouk-HMCTS**

### CODE REVIEW

File: subscription-descriptor-parser/src/main/java/uk/gov/justice/maven/generator/io/files/parser/eventsource/EventSourcesFileParser.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package uk.gov.justice.maven.generator.io.files.parser.eventsource;
+
+import static com.fasterxml.jackson.annotation.JsonCreator.Mode.PROPERTIES;
+import static java.lang.String.format;
+
+import uk.gov.justice.domain.eventsource.EventSource;
+import uk.gov.justice.domain.eventsource.EventSources;
+import uk.gov.justice.maven.generator.io.files.parser.FileParser;
+import uk.gov.justice.maven.generator.io.files.parser.YamlFileValidator;
+import uk.gov.justice.maven.generator.io.files.parser.subscriptiondescriptor.SubscriptionDescriptorException;
```
[the remark this answers is not in the document]
Comment (MohamedFarouk-HMCTS) [reply subscription-descriptor-parser/src/main/java/uk/gov/justice/maven/generator/io/files/parser/eventsource/EventSourcesFileParser.java]: removed

File: subscription-descriptor-parser/src/main/java/uk/gov/justice/maven/generator/io/files/parser/subscriptiondescriptor/SubscriptionDescriptorFileParserFactory.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package uk.gov.justice.maven.generator.io.files.parser.subscriptiondescriptor;
+
+
+import static java.nio.file.Paths.get;
+
+import uk.gov.justice.domain.subscriptiondescriptor.SubscriptionDescriptor;
+import uk.gov.justice.maven.generator.io.files.parser.FileParserFactory;
+import uk.gov.justice.maven.generator.io.files.parser.YamlFileToJsonObjectConverter;
+import uk.gov.justice.maven.generator.io.files.parser.YamlFileValidator;
```
[the remark this answers is not in the document]
Comment (MohamedFarouk-HMCTS) [reply subscription-descriptor-parser/src/main/java/uk/gov/justice/maven/generator/io/files/parser/subscriptiondescriptor/SubscriptionDescriptorFileParserFactory.java]: removed

File: subscription-descriptor-parser/src/main/java/uk/gov/justice/maven/generator/io/files/parser/subscriptiondescriptor/SubscriptionDescriptorFileParserFactory.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+public class SubscriptionDescriptorFileParserFactory implements FileParserFactory<SubscriptionDescriptor> {
+
+    private static final String yamlSchemaFileName = "/json/schema/subscription-schema.json";
```
[the remark this answers is not in the document]
Comment (MohamedFarouk-HMCTS) [reply subscription-descriptor-parser/src/main/java/uk/gov/justice/maven/generator/io/files/parser/subscriptiondescriptor/SubscriptionDescriptorFileParserFactory.java]: renamed

File: subscription-descriptor-parser/src/main/java/uk/gov/justice/maven/generator/io/files/parser/subscriptiondescriptor/SubscriptionDescriptorYamlFileValidator.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
+package uk.gov.justice.maven.generator.io.files.parser.subscriptiondescriptor;
+
+import static java.lang.String.format;
+
+import uk.gov.justice.maven.generator.io.files.parser.YamlFileToJsonObjectConverter;
+import uk.gov.justice.maven.generator.io.files.parser.YamlFileValidator;
+import uk.gov.justice.maven.generator.io.files.parser.eventsource.EventSourcesException;
```
[the remark this answers is not in the document]
Comment (MohamedFarouk-HMCTS) [reply subscription-descriptor-parser/src/main/java/uk/gov/justice/maven/generator/io/files/parser/subscriptiondescriptor/SubscriptionDescriptorYamlFileValidator.java]: removed

File: subscription-descriptor-parser/src/main/java/uk/gov/justice/domain/eventsource/EventSources.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
+package uk.gov.justice.domain.eventsource;
+
+import java.util.List;
+
+public class EventSources {
+
+    private final List<EventSource> eventSources;
```
[the remark this answers is not in the document]
Comment (MohamedFarouk-HMCTS) [reply subscription-descriptor-parser/src/main/java/uk/gov/justice/domain/eventsource/EventSources.java]: suppressed warning as i need this to be called eventSources for jackson pojo mapping

File: subscription-descriptor-parser/src/main/java/uk/gov/justice/maven/generator/io/files/parser/eventsource/EventSourcesFileParserFactory.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+import uk.gov.justice.domain.eventsource.EventSource;
+import uk.gov.justice.maven.generator.io.files.parser.FileParser;
+import uk.gov.justice.maven.generator.io.files.parser.FileParserFactory;
+import uk.gov.justice.maven.generator.io.files.parser.YamlFileToJsonObjectConverter;
+
+import java.util.List;
+
+public class EventSourcesFileParserFactory implements FileParserFactory<List<EventSource>> {
+
+    private static final String yamlSchemaFileName = "/json/schema/event-source-schema.json";
```
[the remark this answers is not in the document]
Comment (MohamedFarouk-HMCTS) [reply subscription-descriptor-parser/src/main/java/uk/gov/justice/maven/generator/io/files/parser/eventsource/EventSourcesFileParserFactory.java]: Suppressed warning "squid:S1075" as I dont want to bring in from configuration file

**Review by mapingo — CHANGES_REQUESTED** [review]
The subscription yaml has been moved to the framework.  The event source changes should be applied to the framework.

**Pull request closed without merging**